lib/rules/aria-allowed-role-matches.js

```diff
@@ -0,0 +1,18 @@
+const { isValidRole, getRole } = axe.commons.aria;
+
+// get role(s) as array(segments), with fallback and dpub
+const roleSegments = getRole(node, {
+	segments: true,
+	fallback: true,
+	dpub: true
+});
+
+// filter invalid roles
+const validRoles = roleSegments.filter(role => isValidRole(role));
```

Comment:
getRole already filters out invalid roles, doesn't it? Couldn't you just do this:

```javascript
return getRole(node, { noImplicit: true, dpub: true }) !== null
```